classes/Product.php

```diff
@@ -5702,7 +5702,7 @@ public static function getProductProperties($id_lang, $row, Context $context = n
         $unitPriceRatio = self::computeUnitPriceRatio($row, $id_product_attribute, $quantity, $context);
         $row['unit_price_ratio'] = $unitPriceRatio;
         $row['unit_price_tax_excluded'] = $unitPriceRatio != 0 ? $priceTaxExcluded / $unitPriceRatio : 0.0;
-        $row['unit_price_tax_included'] = $unitPriceRatio != 0 ? $priceTaxIncluded / $unitPriceRatio : 0.0;
+        $row['unit_price_tax_included'] = $row['unit_price'] * (1 + $row['rate'] / 100);
```

Comment:
Although this fixes the initial problem (which is already enough for the bug) as you correctly stated when we discussed this if the value was incorrect it means the `self::computeUnitPriceRatio` method probably doesn't compute it correctly.

It might be worth looking into it with more details to understand why, maybe it's forgetting the ecotax part so the ecotax value should be added manually. Unless it's a problem related to how `Product::getPriceStatic` is used in `Product::computeUnitPriceRatio` which makes it ignore the ecotax value here https://github.com/lartist/PrestaShop/blob/b1e8cf4591b77a35bdc97c2342f5d6f7715e2883/classes/Product.php#L5758

Reply:
Actually, as we discussed, this fix I suggested is not correct because it always displays the same unit price. I forgot that the unit price is supposed to be adapted when a discount is applied on the product (cart rules by quantity, by cart total sum, ...)

So we should actually compute which percent of discount has been applied to the product in the current context and then apply the same reduction to the unit price. (ex: product price was reduced by 20% then the unit price should also be reduced by 20%)